pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -147,14 +149,35 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 		proxyClientCerts = append(proxyClientCerts, clientCert)
 	}
 
+	// TODO you have to know every APIGroup you're enabling or upstream will panic.  It's alternative to panicing is Fataling
+	// It needs a refactor to return errors
 	storageDestinations := master.NewStorageDestinations()
-	storageDestinations.AddAPIGroup("", databaseStorage)
+	// storageVersions is a map from API group to allowed versions that must include all the valid versions
+	// turns out, this must match values from storageDestinations or the master will fatal out, but
+	// the two are directly linked or validated any way upstream.  Upstream seems to be bugged and only allows
```

Comment:
Do you mean "the two are *not* directly linked or validated in any way upstream" (emphasis mine)